### DISCUSSION TITLE
Prevent the host from sleeping for the duration of a process

### DISCUSSION THREAD

**Discussion by DanEble** [lima-vm/lima#2291] [Q&A]
Is there a command I could run in a Linux VM (vz) that would have the effect of macOS `caffeinate`?  The scenario I would like to address is this:

1. laptop on battery with a fairly short inactivity timeout
2. working in non-GUI, interactive shell in Docker container set up with Colima
3. start a build (`make ...`)
4. lock macOS UI
5. walk away; return later
6. find laptop asleep with build incomplete
 
Ideally, the computer would finish the build and then sleep.  It would be convenient if I could "caffeinate" step 3.

**Comment by jandubois** [comment]
> It would be convenient if I could "caffeinate" step 3.

~Why can't you? Isn't this just `caffeinate -i make`? Does this not work? It is the example straight from `man caffeinate`.~

I missed that you run `make` inside the container and not on the host.

So maybe you can `caffeinate -i colima ssh -- make` from the host?

**Reply by DanEble** [reply]
> maybe you can caffeinate -i colima ssh -- make from the host?

Thanks for your reply.  I could probably make that work, though I would prefer to keep the container workflow from polluting my macOS user's shell history.  I suppose I could handle that by starting a shim shell on the macOS side and directing it to use a history file other than the default; however, if I could equip the container with a command like `caffeinate`, other developers could benefit from it without having to solve the problem in their own way.

I'm curious whether there is a technical reason that a command like `caffeinate` could not work inside the VM, or is it just a matter of finding a motivated developer to create it, or is that unknown?

**Reply by jandubois** [reply]
Any command you run inside the container would only affect the VM and not the host, so could not prevent the host from sleeping.

Now theoretically you could implement something to tell the guest agent in the VM to tell the host agent to prevent the host from sleeping.

I don't think it would be worth the effort, and am generally opposed to add functionality that allows the VM to run code on the host, as it is a potential security vulnerability. You want containers to be isolated and not be able to execute code outside the container, not just in the VM but even on the host.
